Add scheduleCommand

AlexCQY opened the pull request:
- adds `ScheduleCommand`
- adds `ScheduleCommandParser`
- adds unit tests
-> given a  `String` `userInputdate` scheduleCommand outputs the list of students that the tutor has lessons with on that day of the week

- change `dayOfWeek` field in `ClassTime` to Java enum `DayOfWeek`
- Add unit tests for methods in `ClassTime`

ToDo:
-add integration tests for `ScheduleCommand` (Test behavior with `Model`)
-update UG

AlexCQY committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2103T-W15-2/tp into master

AlexCQY committed:
update classTime dayOfWeek

AlexCQY committed:
Add schedule command and parser

AlexCQY committed:
add unit tests for scheduleCommand

AlexCQY committed:
fix checkstyle

AlexCQY committed:
add unit test for ScheduleCommandParser

AlexCQY committed:
fix checkstyle

AlexCQY committed:
add unit tests in parserTests

AlexCQY committed:
fix checkstyl for CI

AlexCQY committed:
Resolve merge conflicts
ClassTime.java
ClassTimeTest.java

hogantan commented on src/main/java/seedu/address/logic/commands/DeleteCommand.java:
```diff
@@ -11,18 +11,18 @@
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
+ * Deletes a student identified using it's displayed index from the address book.
```
Address book could be changed to Reeve here

AlexCQY replied:
Okay thanks!

hogantan commented on src/main/java/seedu/address/logic/commands/ScheduleCommand.java:
```diff
@@ -0,0 +1,58 @@
+public class ScheduleCommand extends Command {
+
+    public static final String COMMAND_WORD = "schedule";
+
```
If we are finding class time with respect to dayofweek, then would that mean that the user input of a specific date is redundant? If we just convert the specific user input date in dayofweek, we could just ask user to input 1,2,3,4,5,6,7 (Integer day of week).

AlexCQY replied:
Hmmm what I thought was that input a date is more flexible and convenient. For instance user might want to look up a his/her schedule on a particular date, our app would translate that date internally to the day of week so that the user wouldnt need to check up the calendar. I was thinking of an extension along this lines too, where user could input a string indicating the day or week. What do you think?

hogantan replied:
Oh that is true also. It is easier for a user to remembers the current date rather than the specific day! If we assume this, then I think the current implementation is fine.

VaishakAnand commented on src/main/java/seedu/address/model/student/admin/ClassTime.java:
```diff
@@ -96,8 +104,9 @@ public String convertClassTimeToUserInputString() {
         return String.format("Day of week: %s, Start time: %s, End Time: %s",
-                this.dayOfWeek,
+                dayDisplayName,
                 this.startTime.format(TIME_FORMATTER),
                 this.endTime.format(TIME_FORMATTER));
```
Need to change `TIME_FORMATTER` to `OUTPUT` for both startTime and endTime

AlexCQY replied:
Okay thanks!

AlexCQY committed:
update classTime toString()

AlexCQY committed:
update java doc to reeve

VaishakAnand reviewed (approved):
LGTM

ong-yinggao98 reviewed (approved):
Looks good, could you just do some tests on your fork tonight? Just test the ScheduleCommand execute method to make sure this works and then raise the PR anytime. In the meantime I'll just approve this since it looks about right.

ong-yinggao98 commented:
Perhaps you could also look into making ClassTime and Student implement Comparable so we can sort students in the `filteredList` in order of who goes first.

hogantan reviewed (approved):
LGTM

AlexCQY committed:
-Resolve conflicts:
#	src/main/java/seedu/address/logic/parser/ReeveParser.java
#	src/test/java/seedu/address/logic/parser/ReeveParserTest.java
# src/test/java/seedu/address/testutil/StudentUtil.java

-Add integration test for scheduleCommand
-Update user guide

AlexCQY committed:
fix checkstyle
